libgit/autogit_manager.go

```diff
@@ -0,0 +1,404 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/eapache/channels"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	billy "gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type resetReq struct {
+	srcTLF     *libkbfs.TlfHandle
+	srcRepo    string
+	branchName string
+	dstTLF     *libkbfs.TlfHandle
+	dstDir     string
+	doneCh     chan struct{}
+}
+
+func (r resetReq) dstPath() string {
+	return path.Clean(path.Join(
+		r.dstTLF.GetCanonicalPath(), r.dstDir, r.srcRepo, r.branchName))
```

Comment:
As you mentioned in [another comment](https://github.com/keybase/kbfs/pull/1406/files#r158985366), `path.Join` already calls `path.Clean` before returning: https://golang.org/src/path/path.go?s=4034:4066#L145